pkg/ccl/backupccl/create_scheduled_backup.go

```diff
@@ -139,56 +201,59 @@ func doCreateBackupSchedule(
 			env = knobs.JobSchedulerEnv
 		}
 	}
-	sj := jobs.NewScheduledJob(env)
-
-	// Prepare arguments for scheduled backup execution.
-	args := &ScheduledBackupExecutionArgs{}
-	// TODO(yevgeniy): Support incremental backup
-	args.BackupType = ScheduledBackupExecutionArgs_FULL
 
-	// Set schedule name; if one was not provided, assign default value
-	if eval.scheduleName != nil {
-		name, err := eval.scheduleName()
-		if err != nil {
-			return err
-		}
-		sj.SetScheduleName(name)
-	} else {
-		fmtCtx := tree.NewFmtCtx(tree.FmtSimple)
-		fmt.Fprintf(fmtCtx, "%s BACKUP OF ", args.BackupType)
-		if eval.Targets == nil {
-			fmtCtx.WriteString("CLUSTER")
-		} else {
-			eval.Targets.Format(fmtCtx)
-		}
-		sj.SetScheduleName(fmtCtx.String())
+	// Evaluate incremental and full recurrence.
+	incRecurrence, err := computeScheduleRecurrence(env.Now(), eval.recurrence)
+	if err != nil {
+		return err
+	}
+	fullRecurrence, err := computeScheduleRecurrence(env.Now(), eval.fullBackupRecurrence)
+	if err != nil {
+		return err
 	}
 
-	if eval.recurrence != nil {
-		recurrence, err := eval.recurrence()
-		if err != nil {
-			return err
-		}
-		err = sj.SetSchedule(recurrence)
-		if err != nil {
-			return err
+	if incRecurrence != nil && fullRecurrence == nil {
+		// It's an enterprise user; let's see if we can pick a reasonable
+		// full  backup recurrence based on requested incremental recurrence.
+		fullRecurrence = pickFullRecurrenceFromIncremental(incRecurrence)
+
+		if fullRecurrence == forceFullBackup {
+			fullRecurrence = incRecurrence
+			incRecurrence = nil
 		}
 	}
 
-	if err := setScheduleOptions(eval, &p.ExtendedEvalContext().EvalContext, sj); err != nil {
-		return err
+	if fullRecurrence == nil {
+		return errors.AssertionFailedf(" full backup recurrence should be set")
 	}
 
-	// Prepare backup statement.
+	// Prepare backup statement (full).
 	backupNode := &tree.Backup{
 		Options: tree.BackupOptions{
 			CaptureRevisionHistory: eval.BackupOptions.CaptureRevisionHistory,
 			Detached:               true,
 		},
+		Nested:         true,
+		AppendToLatest: false,
+	}
+
+	if backupNode.Options.CaptureRevisionHistory && !eval.isEnterpriseUser {
```

Comment:
Ugh, all this duplication of BACKUP's planning is pretty unfortunate. Right now BACKUP planning is too much of a monolith hairball to re-use but it'd be nice if we could TODO pulling e.g. these enterprise checks into a helper and reusing the same code, otherwise this is bound to rot if someone adds a new enterprise check in BACKUP and misses it here.